### PR TITLE
Allow multiple geometric derivatives in single run

### PR DESCRIPTION
**Pull request by stigrj**
Make `geometric_derivative` a vector of properties, similar to the others. This allows you to compute several geometric derivatives with different parameters in a single run. This is done by dry-running `mrchem --dryrun molecule`, which will produce the `geom-1` entry in `molecule.json`, and then adding more entries manually before running explicitly the main executable as `mrchem.x molecule.json`, e.g. with different smoothing parameters:

```
        "geometric_derivative": {
          "geom-1": {
            "operator": "h_nuc_grad",
            "precision": 0.0001,
            "smooth_prec": 0.01
          },
          "geom-2": {
            "operator": "h_nuc_grad",
            "precision": 0.0001,
            "smooth_prec": 0.001
          },
          "geom-3": {
            "operator": "h_nuc_grad",
            "precision": 0.0001,
            "smooth_prec": 0.0001
          },
          "geom-4": {
            "operator": "h_nuc_grad",
            "precision": 0.0001,
            "smooth_prec": 1e-05
          },
          "geom-5": {
            "operator": "h_nuc_grad",
            "precision": 0.0001,
            "smooth_prec": 1e-05
          }
        }
```

### CODE REVIEW

**Comment by stigrj** [comment]
Arg, need to update the gradient test

File: src/driver.cpp [hunk 1]
```diff
@@ -443,6 +445,40 @@ void driver::scf::calc_properties(const json &json_prop, Molecule &mol) {
...
+            GeometricDerivative &G = mol.getGeometricDerivative(id);
+
+            for (auto k = 0; k < mol.getNNuclei(); ++k) {
+                const auto nuc_k = nuclei[k];
+                auto Z_k = nuc_k.getCharge();
+                auto R_k = nuc_k.getCoord();
+                auto smooth = detail::nuclear_gradient_smoothing(mol.getNNuclei(), smoothing, Z_k);
+                NuclearGradientOperator h(Z_k, R_k, smooth);
+                h.setup(prec);
+                G.getNuclear().row(k) = Eigen::VectorXd::Zero(3);
```
Comment (robertodr) [inline src/driver.cpp]: Why not do the `getNuclear` only once outside the double loop? Why zero out the rows when they will be filled later on anyway?

File: src/driver.cpp [hunk 2]
```diff
@@ -443,6 +445,40 @@ void driver::scf::calc_properties(const json &json_prop, Molecule &mol) {
...
+                for (auto l = 0; l < mol.getNNuclei(); ++l) {
+                    if (l == k) continue;
+                    const auto nuc_l = nuclei[l];
+                    auto Z_l = nuc_l.getCharge();
+                    auto R_l = nuc_l.getCoord();
+                    std::array<double, 3> R_kl = {R_k[0] - R_l[0], R_k[1] - R_l[1], R_k[2] - R_l[2]};
+                    auto R_kl_3_2 = std::pow(math_utils::calc_distance(R_k, R_l), 3.0);
+                    G.getNuclear().row(k) -= Eigen::Map<Eigen::RowVector3d>(R_kl.data()) * (Z_k * Z_l / R_kl_3_2);
+                }
+                G.getElectronic().row(k) = h.trace(Phi).real();
```
Comment (robertodr) [inline src/driver.cpp]: Same comment here about the `getElectronic`

File: src/driver.cpp (hunk 1 above)
Comment (robertodr) [reply src/driver.cpp]: If zeroing out is really necessary, you could do `G.getNuclear().row(k) = Eigen::RowVector3d::Zero()` should be an infinitesimal more efficient :smile:

**Review by robertodr — CHANGES_REQUESTED** [review]
I think we can micro-optimize accesses to matrices in the loop

File: src/driver.cpp (hunk 1 above)
Comment (stigrj) [reply src/driver.cpp]: Actually, the tensor is now initialized as `nan` in order to separate "not computed" from "computed but zero", so zeroing out is necessary

**Review by robertodr — APPROVED** [review]

**Pull request merged**